File: MovieServer/MovieServer.API/MovieServer.Shared/src/Domain/Aggregation.ts
import IDomainEvent from "./IDomainEvent";

export default class Aggregation {
    public Version: number = 0;
    private _events: IDomainEvent[] = [];
    public Events: IDomainEvent[] = this._events;

    protected AddEvent(event: IDomainEvent): void {
        this._events.push(event);
    }

    public CreateEvent(): void {
        this._events = []
    }
}

File: OrderServer/OrderServer.API/OrderServer.Shared/Domain/Aggregation.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace OrderServer.Shared.Domain
{
    public class Aggregation
    {
        public Guid Guid { get; set; } = Guid.NewGuid();
        private List<IDomainEvent> _event = new();
        public IEnumerable<IDomainEvent> Events
        {
            get
            {
                return _event;
            }
            private set { }
        }

        protected void AddEvent(IDomainEvent @event)
        {
            _event.Add(@event);
        }

        protected void ClearEvent() => _event.Clear();
    }
}
